tests/fgpyo/test_sequence.py

```diff
@@ -54,3 +54,37 @@ def test_homopolymer(bases: str, expected_hp_len: int) -> None:
 )
 def test_gc_content(bases: str, expected_gc_content: float) -> None:
     assert gc_content(bases) == expected_gc_content
+
+
+@pytest.mark.parametrize(
+    "bases, n, expected_length",
+    [
+        ("", 2, 0),
+        ("", 1, 0),
+        ("A", 1, 1),
+        ("A", 2, 0),
+        ("ACGT", 2, 2),
+        ("AACCGGTT",2,  2),
+        ("TTTTCCCGGA",2,  4),
+        ("ACCGGGTTTT",2,  4),
+        ("GTGTGTAAAA", 2, 6),
+        ("GTGTGTGAAAA",2,  6),
+        ("GTGTGTACACACAC",2,  8),
+        ("GTGTGTGACACACAC",2,  8),
+        ("AAACTCTCTCGG", 2, 6),
+        ("AACTCTCTCGGG", 2, 6),
+        ("TGTATATATA", 2, 8),
+        ("TGTATATATAC",2,  8),
+        ("TGATATATATC",2,  8),
+        ("TGATATATAC", 2, 6),
+        ("TAGTAGTAG", 3, 9),
+        ("TATAGTAGTAGTA", 3, 9),
+        ("TACGTACGTACTG", 4, 8),
+        ("TAGTAGTAGTAG", 3, 12),
+        ("TAGTAGTAGTAG", 6, 12),
+        ("TAGTAGTAGTAG", 12, 12),
```

Comment:
```suggestion
        ("AACCGGTT", 1, 8),
        ("AACCGGTT", 3, 1),
        ("TAGTAGTAGTAG", 5, 5),
        ("TAGTAGTAGTAG", 7, 7),
```

A few more edge cases?